src/features/claims/claim-templates/hooks/use-template-claims.ts

```diff
@@ -0,0 +1,133 @@
+import { useInsecurePersistentState } from '@hedvig-ui/hooks/use-insecure-persistent-state'
+import { ClaimsFiltersType } from 'pages/claims/list/ClaimsListPage'
+import { useEffect, useMemo, useState } from 'react'
+import { useHistory, useLocation } from 'react-router'
+
+const useQuery = () => {
+  const { search } = useLocation()
+
+  return useMemo(() => new URLSearchParams(search), [search])
+}
+
+export interface TemplateFilters {
+  filters: ClaimsFiltersTypeWithName[]
+}
+
+export interface ClaimsFiltersTypeWithName extends ClaimsFiltersType {
+  name: string
+}
+
+interface Templates {
+  templateActive: boolean
+  selectedTemplate?: number
+  localFilter: ClaimsFiltersType
+  templateFilters: TemplateFilters
+  selectTemplate: (id: number) => void
+  createTemplate: (id: number, filter: ClaimsFiltersTypeWithName) => void
+  editTemplate: (filter: ClaimsFiltersType, id?: number) => void
+  editTemplateWithName: (id: number, filter: ClaimsFiltersTypeWithName) => void
+  removeTemplate: (id: number) => void
+}
+
+export const useTemplateClaims = (): Templates => {
+  const filterQuery = useQuery().get('filter')
+
+  const [templateActive, setTemplateActive] = useState(false)
+  const [selectedTemplate, setSelectedTemplate] = useState<number>()
```

Comment:
Is the default supposed to be undefined/null? Because then I think we should be explicit with this

```suggestion
  const [selectedTemplate, setSelectedTemplate] = useState<number | null>(null)
```